### PR TITLE
fix(live-fns/packagist): version detection

### PR DESCRIPTION
**Pull request by jackmu95**
The packagist version detection for `latest` ist currently broken when a package published a patch version for a older minor release. Than this patch version is detected as the latest release (e.g.`v1.0.0` => `v2.0.0` => `v1.0.1`).

This PR adds a `compareVersions` sort function based on [compare-versions](https://github.com/omichelsen/compare-versions) to sort the versions returned from the packagist API before detecting the latest.

Other changes:
- `nodev` filters out every version containing `dev` (e.g. `1.0.x-dev`)
- `pre` filters out versions starting with `dev` (e.g. `dev-master`)
- Some minor code improvements

### CODE REVIEW

File: libs/utils/compare-versions.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const indexOrEnd = (str, q) => str.indexOf(q) === -1 ? str.length : str.indexOf(q)
```
Comment (amio) [inline libs/utils/compare-versions.js]: Better use `compare-versions` directly other than copy it's code here

Comment (jackmu95) [reply libs/utils/compare-versions.js]: The problem is that it throws an exception when a invalid semver version is detected and they are quit common on packagist. This was the only reason for not using it directly.

Comment (amio) [reply libs/utils/compare-versions.js]: Hmm, indeed a problem, but force comparing may cause unexpected exceptions too. What about filter versions with validation before compare?

Comment (jackmu95) [reply libs/utils/compare-versions.js]: `semver` event provides it's own sorting function which would be ideal.
Maybe we need to sort out invalid versions first, sort them separately and put the correctly sorted (and semver valid) versions afterwards?

Comment (jackmu95) [reply libs/utils/compare-versions.js]: @amio I will try to come up with a initial solution later today.

Comment (amio) [reply libs/utils/compare-versions.js]: Sure, no hurry ☕️

**Comment by jackmu95** [comment]
@amio Now the sort priority is following:
1. Non-numeric versions (e.g. `dev-master`)
2. Numeric, non-semver versions (e.g. `1.0.x-dev`)
3. Semver versions (e.g. `1.0.0`)

File: libs/utils/compare-versions.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+module.exports = (v1, v2) => {
+  const validV1 = semver.valid(v1)
+  const validV2 = semver.valid(v2)
+
+  const isV1Valid = !!validV1
+  const isV2Valid = !!validV2
+
+  if (!isV1Valid && !isV2Valid) {
+    const numberVersionRegex = /^\d/
```
Comment (amio) [inline libs/utils/compare-versions.js]: Seems this can be omitted, since `semver.coerce()` could turn `numberVersion` into valid semver.

**Comment by jackmu95** [comment]
@amio `semver.coerce()` seems like the perfect solution :)

**Comment by amio** [comment]
Nice, merge it!

**Pull request merged**